chutney/ui/src/app/core/services/login.service.ts

```diff
@@ -27,20 +28,33 @@ export class LoginService {
 
   constructor(
     private http: HttpClient,
-    private router: Router
+    private router: Router,
+    private ssoOpenIdConnectService: SsoOpenIdConnectService
   ) { }
 
-  initLogin(url?: string) {
-    this.currentUser(true).pipe(
-        tap(user => this.setUser(user))
-    ).subscribe(
-        () => this.navigateAfterLogin(url),
-        () => {
-            const nextUrl = this.nullifyLoginUrl(url);
-            const queryParams: Object = isNullOrBlankString(nextUrl) ? {} : { queryParams: { url: nextUrl } };
-            this.router.navigate(['login'], queryParams);
-        }
-    );
+  initLogin(url?: string, headers: HttpHeaders | {
```

Comment:
use  `headers?` instead of initialization.
Is there any way to have more simple Type than `HttpHeaders | {[header: string]: string | string[]; }` ?